Source/Core/Core/AchievementManager.cpp

```diff
@@ -814,8 +988,7 @@ AchievementManager::ResponseType AchievementManager::VerifyCredentials(const std
   return r_type;
 }
 
-AchievementManager::ResponseType
-AchievementManager::ResolveHash(std::array<char, HASH_LENGTH> game_hash)
+AchievementManager::ResponseType AchievementManager::ResolveHash(Hash game_hash, u32* game_id)
```

Comment:
A bit tangential, but:

```suggestion
AchievementManager::ResponseType AchievementManager::ResolveHash(const Hash& game_hash, u32* game_id)
```

would avoid an unnecessary copy.

Also maybe just use a struct for the return value instead of an out pointer, since you can just do:

```cpp
const auto [response, new_game_id] = ResolveHash(current_hash);
```

which would be a little more straightforward and make the sanitizing behavior completely handled by the function, rather than relying on initializing the value at the point of the call (i.e. consider a refactor, where the initialization for the variable pointed to gets removed. with the current code, this would cause an uninitialized read in one path, whereas a return value prevents that from happening entirely)

Reply:
Is this a change you feel I should make across the board? Because that's a lot and probably should go in a separate PR if so.

Reply:
@lioncash Would still like your opinion on this, many thanks.